### PR TITLE
Offer test suite packages on SSH minions

### PR DESCRIPTION
**Pull request by Bischoff**
"SSH minion" clients are managed with the generic "host" module and no role.

They however differ from a generic host in the sense that they need to be able to benefit from test suite's repository and packages. This is what this PR adds.

### CODE REVIEW

File: salt/client/testsuite.sls [hunk 1]
```diff
@@ -18,12 +18,6 @@ client_cucumber_requisites:
 
 {% if grains['os'] == 'SUSE' %}
 
-testsuite_build_repo:
-  file.managed:
-    - name: /etc/zypp/repos.d/Devel_Galaxy_BuildRepo.repo
```
Comment (moio) [inline salt/client/testsuite.sls]: Just to double check: is this really never needed for traditionally-managed clients?

File: salt/default/testsuite.sls [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{% if grains.get('testsuite') | default(false, true) %}
+{% if grains.get('role') == '' %}
```
Comment (moio) [inline salt/default/testsuite.sls]: Shouldn't this be `is None`?

Comment (Bischoff) [reply salt/default/testsuite.sls]: i looked at the doc of get() and they said that if default value is absent, then return value is empty string. However i might be wrong and it's not tested yet. Feel free to object stronger ;-) .

File: salt/client/testsuite.sls (hunk 1 above)
Comment (Bischoff) [reply salt/client/testsuite.sls]: traditional clients are also "default" hosts.

.... argh! I test for no role then! *bug*, thanks

File: salt/default/testsuite.sls (hunk 2 above)
Comment (mateiw) [reply salt/default/testsuite.sls]: Hmm, maybe `grains.get('role', '') == ''` would be better. Afaik `get(key)` without a default value throws an err if the key is missing.

File: salt/client/testsuite.sls (hunk 1 above)
Comment (Bischoff) [reply salt/client/testsuite.sls]: ok, fixing by doing a permissive clause as you suggested in irc

File: salt/default/testsuite.sls (hunk 2 above)
Comment (Bischoff) [reply salt/default/testsuite.sls]: I've put a | default() guard cause anyway... Let's see if that works...

File: salt/default/testsuite.sls [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{% if grains.get('testsuite') | default(false, true) %}
+{% if (grains.get('role') | default('ssh_minion', true)) in ['client', 'minion', 'ssh_minion'] %}
```
Comment (moio) [inline salt/default/testsuite.sls]: `grains.get('role') in ['client', 'minion', None]` should work

Comment (Bischoff) [reply salt/default/testsuite.sls]: Yes, but i find this more expressive so, as it explicitely mention a pseudo-role "ssh-minion".

If you insist I'll change.

**Comment by Bischoff** [comment]
Full scale testsuite test is running currently.

In last version, I had to move "testsuite" grain to "host" top-level module in openstack and libvirt providers. Having a real "ssh_minion" role would have avoided that :smile_cat: .

File: salt/default/testsuite.sls (hunk 3 above)
Comment (Bischoff) [reply salt/default/testsuite.sls]: That's precisely the problem with not having a dedicated role... I'm not sure this should apply to any vanilla minion... anyway. I'll change it.

**Comment by Bischoff** [comment]
Tests were green. OK to merge?

**Comment by Bischoff** [comment]
Done, thanks.

**Pull request merged**